solana/program_v2/programs/gateway_v2/src/instructions/admin/create_network.rs

```diff
@@ -1,42 +1,20 @@
 use anchor_lang::prelude::*;
 
 use crate::errors::NetworkErrors;
-use crate::state::{
-    AuthKey, GatekeeperNetwork, NetworkFeesPercentage, NetworkKeyFlags, SupportedToken,
-};
+use crate::state::{AuthKey, GatekeeperNetwork, NetworkFeesPercentage, SupportedToken};
+use crate::util::check_network_auth_threshold;
 
 pub fn create_network(ctx: Context<CreateNetworkAccount>, data: CreateNetworkData) -> Result<()> {
     let network = &mut ctx.accounts.network;
     let authority = &ctx.accounts.authority;
 
-    // Check there are auth_keys provided (TODO: Is this necessary? The next check implies this)
-    if data.auth_keys.is_empty() {
-        return Err(error!(NetworkErrors::NoAuthKeys));
-    }
-
-    // Check if there are enough auth_keys with the AUTH flag set
-    // TODO: Move this check into a trait OR helper function and verify with q require OR in a constraint.
-    if data
-        .auth_keys
-        .iter()
-        .filter(|key| {
-            NetworkKeyFlags::from_bits_truncate(key.flags).contains(NetworkKeyFlags::AUTH)
-        })
-        .count()
-        < data.auth_threshold as usize
-    {
-        return Err(error!(NetworkErrors::InsufficientAuthKeys));
-    }
-
     network.auth_threshold = data.auth_threshold;
```

Comment:
I'd move these assignments into some kind of initializer method on the network.

Reply:
Cool, will do that :)